db_stress_tool/db_stress_shared_state.h

```diff
@@ -260,10 +281,14 @@ class SharedState {
   // This is useful for crash-recovery testing when the process may crash
   // before updating the corresponding expected value
   //
-  // Requires external locking covering `key` in `cf` to prevent concurrent
-  // write or delete to the same `key`.
-  PendingExpectedValue PreparePut(int cf, int64_t key) {
-    return expected_state_manager_->PreparePut(cf, key);
+  // It can fail and `*prepared` will be set to false if the previous write or
```

Comment:
Do we still need these changes with the loop-retry approach?

Reply:
I intend to have this so we don't have to answer what value read is expected to return when we see a pending write and pending delete to the same key. This is possible because we release lock during sleep before retrying for write to the same key. I sense it will be too much reasoning complexity to add to Expected State for an edge case.